HACKING.md

```diff
@@ -0,0 +1,99 @@
+# Building barnabas
+
+Barnabas is using `make` as its main build system. Our make build supports 
+several different targets mainly for building and pushing docker images.
+
+<!-- TOC depthFrom:2 -->
+
+- [Docker images](#docker-images)
+    - [Building Docker images](#building-docker-images)
+    - [Tagging and pushing Docker images](#tagging-and-pushing-docker-images)
+- [Building everything](#building-everything)
+- [Release](#release)
+
+<!-- /TOC -->
+
+## Docker images
+
+### Building Docker images
+
+The `docker_build` target will build the Docker images provided by the 
+Barnabas project. You can build all Barnabas Docker images by calling 
+`make docker_build` from the root of the Barnabas repository. Or you can build 
+an individual Docker image by running `make docker_build` from the 
+subdirectories with their respective Dockerfiles - e.g. `kafka_base`, 
+`kafka_statefulsets` etc.
+
+The `docker_build` target will always build the images under the 
+`enmasseproject` organization. This is necessary in order to be able to reuse 
+the base image you might have just built without modifying all Dockerfiles.
+
+To configure the `docker_build` target you can set environment variables:
+* `DOCKER_ORG` should contain the Docker organization for tagging/pushing 
+  the images (default is your username)
+* `DOCKER_TAG` configured Docker tag (default is `latest`)
+
+### Tagging and pushing Docker images
+
+Target `docker_tag` can be used to tag the Docker images built by the 
+`docker_build` target. This target is automatically called by the `docker_push` 
+target and doesn't have to be called separately. The `docker_push` target will 
+login to the configured container repository and push the images.
+
+To configure the `docker_tag` and `docker_push` targets you can set following 
+environment variables:
+* `DOCKER_ORG` configures the Docker organization for tagging/pushing the 
+  images (defaults to the value of the `$USER` environment variable)
+* `DOCKER_TAG` configured Docker tag (default is `latest`)
+* `DOCKER_REGISTRY` configures the Docker registry where the image will 
+  be pushed (default is `docker.io`)
+
+## Building everything
+
+`make all` command can be used to triger all the tasks above - build the 
+Docker images, tag them and push them to the configured repository.
+
+## Pushing images to the cluster's Docker repo
+
+When developing locally you might want to push the docker images to the docker
+repository running in your local OpenShift cluster. This can be quicker than
+pushing to dockerhub and works even without a network connection.
+
+1. Make sure your cluster is running,
+
+        oc cluster up
+
+   and that you're logged in as `developer`,
+
+        oc whoami
+    
+   If you're not logged in as `developer`, then do so:
+   
+        oc login -u developer
+        
+2. Log in to the Docker repo running in the local cluster:
+
+        docker login -u myproject -p `oc whoami -t` 172.30.1.1:5000
```

Comment:
what if someone is using a different project ? Maybe we should be more general on the project name ?

Reply:
I think any particular name would be too specific. We could use something that's more obviously a placeholder, like `$OC_PROJECT`, but then we have to define what that means (and explain that it's not an env var that means anything in the rest of the Makefiles,, scripts etc, but it just being used to explain). Really what people want when looking at a document like this is something they can copy and paste into a terminal. So overall, I think that would be worse than using a fairly obvious dummy value, like we are here. I'll add a sentence at the start explaining that `myproject` and `developer` could be replaced by appropriate values.

Reply:
What about using `oc project -q` to get the current project?